src/internet_identity/src/storage.rs

```diff
@@ -340,6 +343,23 @@ impl<M: Memory> Storage<M> {
                  (max {max_entries} entries)"
             ));
         }
+
+        // restrict further if II has users to protect existing anchors
+        if self.header.num_anchors > 0 {
+            if self.header.id_range_lo != lo {
+                trap(&format!(
+                    "set_anchor_number_range: specified range [{lo}, {hi}) does not start from the same number ({}) \
+                     as the existing range thus would make existing anchors invalid"
+                    , {self.header.id_range_lo}));
+            }
+            if (hi - lo) < self.header.num_anchors as u64 {
```

Comment:
it would be enough to check `hi >= self.header.id_range_hi` here, right? or am I missing something?

Reply:
Hm, I don't think it would check for the same thing?
* `(hi - lo) < self.header.num_anchors` checks that all existing anchors would still fit (but allows shrinking the range).
* checking for `hi >= self.header.id_range_hi` would only allow growing the range.

I'm happy to change it though, if you feel it makes things clearer.

Reply:
Sorry, I meant in the context of the earlier check! I assumed you were checking that (1) existing anchors are not shifted (by making sure `lo_before` = `lo_after`) and that (2) all anchors still fit (and since `lo_after == lo_before`, this would mean checking that `hi_after >= hi_before`).

But maybe you're checking something else here?

Reply:
But your proposed check number (2) does not actually do that: I.e. `hi_before` does not correspond to the highest _existing_ anchor number, but to the highest _allowed_ anchor number. I.e. if we have the range `(0, 10]` and 5 anchors it is ok to change the range down to `(0, 6]` since the anchors `6, 7, 8, 9` are unused. But maybe this is unnecessary complexity...

Reply:
Oh my, you're right! Can you add a comment about this?